server/punishments.ts

```diff
@@ -679,6 +679,12 @@ export const Punishments = new class {
 		const ipStr = typeof user !== 'string' ? ` [${(user as User).latestIp}]` : '';
 		const roomid = typeof room !== 'string' ? (room as Room).roomid : room;
 		Rooms.global.modlog(`(${roomid}) AUTO${namelock ? `NAME` : ''}LOCK: [${userid}]${ipStr}: ${reason}`);
+
+		const roomObject = Rooms.get(room);
+		const userObject = Users.get(user);
+		if (roomObject.battle && userObject && userObject.connections[0]) {
```

Comment:
Not checking if `roomObject` exists (the return type for getRoom is wrong) makes me nervous, but I think it's always going to be fine in practice.

Reply:
Well, it's going to need a TS override now that we merged your PR. :p